libs/ts-rest/core/src/lib/client.spec.ts

```diff
@@ -541,25 +554,7 @@ describe('custom api', () => {
     await customClient.posts.getPost({
       params: { id: '1' },
```

Comment:
We have a weird pre-existing bug here. For some reason when a custom client is used, all args become optional and args itself becomes optional. In this case, `params` should not be optional but the type for `getPost` is

`getPost: (args?: ({headers?: {"x-test"?: string | undefined, "base-header"?: string | undefined, "x-api-key"?: string | undefined} | undefined, params?: {id: string} | undefined, uploadProgress?: ((progress: number) => void) | undefined} | undefined)) => Promise<...>`

Reply:
I just played with this, I can replicate it, it looks like the "AreAllPropertiesOptional" work doesn't work correctly :(